tools/jlink-nuttx.c

```diff
@@ -72,7 +72,7 @@ enum symbol_e
   NSYMBOLS
 };
 
-__attribute__ ((packed)) struct tcbinfo_s
+struct __attribute__ ((packed)) tcbinfo_s
```

Comment:
Let's do either
```
struct tcbinfo_s __attribute__ ((packed))
```
or
```
struct tcbinfo_s
{
} __attribute__ ((packed));
```
I prefer second variant.
Also seems that 
```
  union
  {
    uint8_t  u[8];
    uint16_t *p;
} reg_off;
```
is missing packed attribute as well.
It should be
```
struct tcbinfo_s
{
  uint16_t pid_off;
  uint16_t state_off;
  uint16_t pri_off;
  uint16_t name_off;
  uint16_t reg_num;
  union
  {
    uint8_t  u[8];
    uint16_t *p;
  } __attribute__ ((packed)) reg_off;
  uint16_t reg_offs[0];
} __attribute__ ((packed));
```

Reply:
Thanks for the feedback, Petro.  I modified the patch according to your proposal.

Reply:
@pkarashchenko is it packed on reg_off really needed? BTW, The device side don't have this yet.